### PR TITLE
Workflows backend filtering/sorting - UX

### LINKED ISSUE #564: Workflows main page refactoring - workflows datagrid - backend filtering and pagination
Workflows main page refactoring - workflows datagrid - backend filtering and pagination

- [x] **Backend**: Implement search rest endpoint 
- [x]  **Frontend**: Paginate datagrid using rest endpoint

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements: #564

### CODE REVIEW

File: ui/src/app/components/workflows/workflows-home/workflows-home.component.html [hunk 1]
```diff
@@ -218,5 +218,12 @@ <h3 class="modal-title">Upload workflow</h3>
       {{workflow.isActive ? 'Yes' : 'No'}}
     </clr-dg-cell>
   </clr-dg-row>
+  <clr-dg-footer>
+    <clr-dg-pagination #pagination [clrDgPageSize]="pageSize" [clrDgPage]="page"
+                       [clrDgTotalItems]="total"></clr-dg-pagination>
+    <clr-dg-page-size [clrPageSizeOptions]="[100,150,200,250,300]">Workflows per page</clr-dg-page-size>
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflows-home/workflows-home.component.html]: Maybe also add 50

File: ui/src/app/stores/workflows/workflows.reducers.ts [hunk 2]
```diff
@@ -487,7 +518,7 @@ export function workflowsReducer(state: State = initialState, action: WorkflowsA
       return {
         ...state,
         loading: false,
-        workflows: [...state.workflows, ...action.payload],
+        // workflows: [...state.workflows, ...action.payload],
```
Comment (kevinwallimann) [inline ui/src/app/stores/workflows/workflows.reducers.ts]: What is this?

Comment (jozefbakus) [reply ui/src/app/stores/workflows/workflows.reducers.ts]: Fixed

File: ui/src/app/components/workflows/workflows-home/workflows-home.component.html (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflows-home/workflows-home.component.html]: Done

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**